libgalois/include/katana/PODVector.h

```diff
@@ -145,15 +177,25 @@ class PODVector {
   void shrink_to_fit() {
     if (size_ == 0) {
       if (data_ != NULL) {
+        if (pinned_) {
+          munlock(data_, capacity_ * sizeof(_Tp));
+        }
         free(data_);
         data_ = NULL;
         capacity_ = 0;
       }
     } else if (size_ < capacity_) {
+      if (pinned_) {
+        munlock(data_, capacity_ * sizeof(_Tp));
+      }
       capacity_ = std::max(size_, kMinNonZeroCapacity);
-      _Tp* new_data_ = static_cast<_Tp*>(
-          realloc(reinterpret_cast<void*>(data_), capacity_ * sizeof(_Tp)));
+      const size_t new_bytes = capacity_ * sizeof(_Tp);
+      _Tp* new_data_ =
+          static_cast<_Tp*>(realloc(reinterpret_cast<void*>(data_), new_bytes));
       KATANA_LOG_DEBUG_ASSERT(new_data_);
+      if (pinned_) {
+        mlock(new_data_, new_bytes);
```

Comment:
Please handle errors in both `mlock` and `munlock`, throughout this code. It might be reasonable to just print a warning (to alert people of the performance issue), but the error shouldn't just be ignored. We may also need to prevent `munlock` from being called if `mlock` fails.